### ISSUE TITLE
Add the host's primary IP to Logcollector's output format

### ISSUE THREAD

**Issue by vikman90** [wazuh/wazuh#4341]
|Depends on|Working branch|Documentation|
|---|---|---|
|#4334|[4341-logcollector-ip](https://github.com/wazuh/wazuh/compare/4341-logcollector-ip)|https://github.com/wazuh/wazuh-documentation/pull/2098|

We need Logcollector to insert the host's IP into the logs via the `<out_format>` option.

## New parameters

```xml
<localfile>
  <log_format>syslog</log_format>
  <location>/var/log/syslog</location>
  <out_format>source: $(hostname)@$(host_ip), log: $(log)</out_format>
</localfile>
```

|Parameter|Description|
|---|---|
|**host_ip**|Host's primary IP.|

## How to get the host's IP

Imitate the agent's behavior:
- Query the control module (_wazuh-modulesd_) on UNIX.
- Call `get_win_agent_ip()` on Windows.

## Data flow

1. The file reloader thread shall get the host's IP and store it into a shared variable.
2. The data sender threads shall read that shared variable, if available.

## Alternatives

There are a few conditions that might prevent Logcollector from getting the host's IP:

1. Modulesd is not running.
2. Network error.
3. Unsupported platform.

In this case, the alternative value of the IP shall be:
```
0.0.0.0
```

## Platforms

This feature must work on the following platforms at least:
- [x] Linux on x86.
- [x] macOS.
- [x] Windows.
- [ ] Solaris on x86.
- [ ] Solaris on SPARC.

In addition, these platforms are desirable:
- [ ] FreeBSD.
- [ ] AIX.
- [ ] HP-UX.
- [ ] Linux on ARM (Raspberry Pi).

**Comment by vikman90** [comment]
## Get host IP address on Solaris

Solaris 11 can handle Linux-like `getifaddrs()` function, but Solaris 10 cannot.

Here is a legacy way to discover all the network interfaces on Solaris 10 using `ioctl()`, and using the first active non-loopback interface as the primary one:

```c
#include <sys/ioctl.h>
#include <net/if.h>
#include <netinet/in.h>
#include <assert.h>

#include <errno.h>
#include <string.h>
#include <stdio.h>
#include <arpa/inet.h>
#include <unistd.h>
#include <stdlib.h>
#include <net/if.h>
#include <sys/sockio.h>

int get_if_num() {
    int fd = socket(AF_INET, SOCK_DGRAM, 0);

    if (fd == -1) {
        return -1;
    }

    struct lifnum ifn = { .lifn_family = AF_INET };

    int retval = ioctl(fd, SIOCGLIFNUM, &ifn);
    close(fd);

    if (retval == -1) {
        return -1;
    }

    return ifn.lifn_count;
}

char * get_primary_ip() {
    char * primary_ip = NULL;

    // Get number of interfaces

    int if_count = get_if_num();

    if (if_count == -1) {
        return NULL;
    }

    // Initialize configuration structure

    struct lifconf if_conf = { .lifc_family = AF_INET, .lifc_len = if_count * sizeof(struct lifreq) };
    if_conf.lifc_buf = malloc(if_conf.lifc_len);
    assert(if_conf.lifc_buf != NULL);

    // Create helper socket

    int fd = socket(AF_INET, SOCK_DGRAM, 0);

    if (fd == -1) {
        goto end;
    }

    // Get interfaces

    if (ioctl(fd, SIOCGLIFCONF, &if_conf) == -1) {
        goto end;
    }

    // Scan interfaces

    int i;
    for (i = 0; i < if_count; i++) {
        struct lifreq * if_req = if_conf.lifc_req + i;

        // Get flags

        if (ioctl(fd, SIOCGLIFFLAGS, if_req) == -1) {
            goto end;
        }

        // Get the first interface that is up and is not loopback

        int flags = if_req->lifr_flags;

        if ((flags & IFF_UP) && (flags & IFF_LOOPBACK) == 0) {
            // Get IP address

            if (ioctl(fd, SIOCGLIFADDR, if_req) == -1) {
                goto end;
            }

            struct sockaddr_in * addr = (struct sockaddr_in *)&if_req->lifr_addr;
            primary_ip = strdup(inet_ntoa(addr->sin_addr));
            break;
        }
    }

end:
    if (fd != -1) {
        close(fd);
    }

    free(if_conf.lifc_buf);
    return primary_ip;
}

int main() {
    char * primary_ip = get_primary_ip();
    assert(primary_ip != NULL);

    printf("Primary IP: %s\n", primary_ip);
    free(primary_ip);

    return EXIT_SUCCESS;
}
```

**Comment by chemamartinez** [comment]
Merged: #4380
